hasona23.ShiftsLogger/ShiftsLoggerCLI/InputHandler.cs

```diff
@@ -0,0 +1,248 @@
+﻿using System.Globalization;
+using System.Text.RegularExpressions;
+using ShiftsLoggerCLI.Models;
+using Spectre.Console;
+
+namespace ShiftsLoggerCLI;
```

Comment:
I think your namespaces should be:
`ShiftsLogger.Api`
and
`ShiftsLogger.Cli`

That way they both share the root `ShiftsLogger` namespace, rather than both being totally independent.

A lot of your validation logic is duplicated and could have existed inside a `ShiftsLogger.Domain` namespace and referenced by both API and CLI projects.